pkg/operator/ceph/csi/secrets.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/config/keyring"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	csiKeyringRBDProvisionerUsername = "client.csi-rbd-provisioner"
+	csiKeyringRBDNodeUsername        = "client.csi-rbd-node"
+	csiRBDNodeSecret                 = "rook-csi-rbd-node-stage"
+	csiRBDProvisionerSecret          = "rook-csi-rbd-provisioner"
+)
+
+const (
+	csiKeyringCephFSProvisionerUsername = "client.csi-cephfs-provisioner"
+	csiKeyringCephFSNodeUsername        = "client.csi-cephfs-node"
+	csiCephFSNodeSecret                 = "rook-csi-cephfs-node-stage"
+	csiCephFSProvisionerSecret          = "rook-csi-cephfs-provisioner"
+)
+
+func createCSIKeyringRBDNode(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringRBDNodeUsername, cephCSIKeyringRBDNodeCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createCSIKeyringRBDProvisioner(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringRBDProvisionerUsername, cephCSIKeyringRBDProvisionerCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createCSIKeyringCephFSNode(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringCephFSNodeUsername, cephCSIKeyringCephFSNodeCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createCSIKeyringCephFSProvisioner(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(csiKeyringCephFSProvisionerUsername, cephCSIKeyringCephFSProvisionerCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func cephCSIKeyringRBDNodeCaps() []string {
+	return []string{
+		"mon", "profile rbd",
+		"osd", "profile rbd",
+	}
+}
+
+func cephCSIKeyringRBDProvisionerCaps() []string {
+	return []string{
+		"mon", "profile rbd",
+		"mgr", "allow rw",
+		"osd", "profile rbd",
+	}
+}
+
+func cephCSIKeyringCephFSNodeCaps() []string {
+	return []string{
+		"mon", "allow r",
+		"osd", "allow rw tag cephfs data=*",
+		"mds", "allow rw",
+	}
+}
+
+func cephCSIKeyringCephFSProvisionerCaps() []string {
+	return []string{
+		"mon", "allow r",
+		"mgr", "allow rw",
+		"osd", "allow rw tag cephfs meta=*",
+	}
+}
```

Comment:
CephFS caps look good to me!

Reply:
Gah, this should be `metadata=*`.

Reply:
I should have caught that... @travisn ^

Reply:
Oh... you guys just changed that to see if it would help. Ya this should definitely be `metadata=*`. Let me see if I can figure out what's going on.

Reply:
There appears to be a bug in the cap handling here. Use this instead:

` 		"osd", "allow rw tag cephfs *=*",`

Reply:
to be clear, should the caps for the cephfs provisioner (line 106) be `metadata=*` and the cephfs node (line 97) be `data=*`? Or should they both be `metadata=*`?

Reply:
Ahhhhhh, it's actually:

`"osd", "allow rw tag cephfs metadata=*",`